inspect_byproducts.py

```diff
@@ -0,0 +1,168 @@
+  Sachit Malik <i.sachitmalik@gmail.com>
+
+<Started>
+  June 6, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Requires>
+  in-toto
+
+<Purpose>
+  Inspections constitute an important part of in-toto. The script take the path
+  of a link file, a random string and an operator, which is used to compare
+  a certain property of the link file with the random string.
+
+  Suppose the link file is located at /user/abc/def/package.45gh325.link
+  and the user wants to check whether for the corresponding step (hence the
+  link file), the corresponding stderr field contains the string "test".
+
+  The usage would be as follows:
+  python inspect_byproducts.py -l  /user/abc/def/package.45gh325.link  -t
+    stderr -o contains test
+
+  General usage:
+  python inspect_byproducts.py -l  <path/to/link/metadata/file>  -t
+    <stdout|stderr> -o [ is | is not | contains | contains not] <string to
+    be tested>
+"""
+
+import os
+import sys
+import argparse
+import in_toto.log
+from in_toto.models.link import Link as link_import
+import securesystemslib.exceptions
+
+
+def inspect_byproducts(link, type, operator, input_string):
+    """
+    <Purpose>
+    A function which performs the inspection as described above depending on
+    various arguments.
+
+    <Arguments>
+      link:
+        the path to the link file
+
+      std:
+        whether to check stdout or stderr field
+
+      operator:
+        is | is not | contains | contains not
+
+      input_string:
+        the string to be checked
+
+    <Exceptions>
+      Raises KeyError, in case the field corresponding to the key
+      in the dictionary is empty
+
+    <Returns>
+      Boolean
+    """
+
+    imported_link = link_import.read_from_file(link)
+
+    std_out_err = imported_link.byproducts[type]
+
+    if operator == 'is':
+      if std_out_err == input_string:
+        return True
+
+    elif operator == 'is-not':
+      if std_out_err != input_string:
+        return True
+
+    elif operator == 'contains':
+      if std_out_err.find(input_string) != -1:
+        return True
+
+    elif operator == 'contains-not':
+      if std_out_err.find(input_string) == -1:
+        return True
+    else:
+      raise('Invalid Operator ' + operator + '. Valid operators: is | is-not |'
+        ' contains | contains-not')
+
+    return False
+
+
+def parse_args():
+    """
+    <Purpose>
+      A function which parses the user supplied arguments.
+
+    <Arguments>
+      None
+
+    <Exceptions>
+      None
+
+    <Returns>
+      Parsed arguments (args object)
+    """
+    parser = argparse.ArgumentParser(
+        description="Inspects the byproducts of a step")
+
+    lpad = (len(parser.prog) + 1) * " "
+
+    parser.usage = ("\n"
+                    "%(prog)s --link <path to link metadata>\n{0}"
+                    "[--type (--stdout | --stderr)]\n{0}"
+                    "[--operator (--is | --is-not | --contains | --"
+                    "contains-not)]\n{0}"
```

Comment:
Another possibility is that you can just drop setting `usage` and have argparse format it for you:
```
(toto-python2) [santiago@LykOS in-toto-inspections]$ python inspect_byproducts.py                                      
usage: inspect_byproducts.py [-h] -l <Path to link metadata> -t                                                        
                             {stdout,stderr} -o            
                             {is,is-not,contains,contains-not}                                                         
                             <String to compare>   
```
To do this, I just did the following:

```diff 

--- a/inspect_byproducts.py
+++ b/inspect_byproducts.py
@@ -114,18 +114,19 @@ def parse_args():
 
     lpad = (len(parser.prog) + 1) * " "
 
-    parser.usage = ("\n"
-                    "%(prog)s --link <path to link metadata>\n{0}"
-                    "[--type (--stdout | --stderr)]\n{0}"
-                    "[--operator (--is | --is-not | --contains | --"
-                    "contains-not)]\n{0}"
-                    "string\n\n"
-                    .format(lpad))
+#   parser.usage = ("\n"
+#                   "%(prog)s --link <path to link metadata>\n{0}"
+#                   "[--type (--stdout | --stderr)]\n{0}"
+#                   "[--operator (--is | --is-not | --contains | --"
+#                   "contains-not)]\n{0}"
+#                   "string\n\n"
+#                   .format(lpad))
 
     in_toto_args = parser.add_argument_group("in-toto-inspection options")
 
     in_toto_args.add_argument("-l", "--link", type=str, required=True,
-                              help="Path to the link file to be inspected")
+                              help="Path to the link file to be inspected",
+                              metavar="<Path to link metadata>")
 
     in_toto_args.add_argument("-t", "--type", choices=['stdout', 'stderr'],
                               type=str, required=True, help="Type of "
@@ -139,7 +140,8 @@ def parse_args():
 
     in_toto_args.add_argument("string", type=str,
                               help="The string to compare with the specified "
-                              "byproduct in the specified link file")
+                              "byproduct in the specified link file",
+                              metavar="<String to compare>")
```